core/src/main/java/com/dtp/core/context/BaseNotifyCtx.java

```diff
@@ -16,12 +17,15 @@ public class BaseNotifyCtx {
 
     private ExecutorWrapper executorWrapper;
 
+    private CapturedExecutor capturedExecutor;
+
     private NotifyItem notifyItem;
 
     public BaseNotifyCtx() { }
 
     public BaseNotifyCtx(ExecutorWrapper wrapper, NotifyItem notifyItem) {
         this.executorWrapper = wrapper;
+        this.capturedExecutor = new CapturedExecutor(wrapper.getExecutor());
```

Comment:
The thread pool data is already snapshot when entering the alarm chain, so can we remove the ExecutorWrapper
I think we can use Prototype design to make a deep copy of ExecutorWrapper

Reply:
For the thread pool object, because it contains the status information of the thread pool and resources such as threads in the thread pool, it may be difficult to perform a full deep copy, or to implement a deep copy still needs to copy all the parameters.
This pr is equivalent to using a partial copy, that is, copying the parameters that affect the display of the alarm information. 
The first question is related to whether to implement a full deep copy. If it is implemented, the ExecutorWrapper can be removed. If the solution is a partial deep copy, then it cannot be removed.

Reply:
> > The thread pool data is already snapshot when entering the alarm chain, so can we remove the ExecutorWrapper I think we can use prototype design to make a deep copy of ExecutorWrapper
> 
> For the thread pool object, because it contains the status information of the thread pool and resources such as threads in the thread pool, it may be difficult to perform a full deep copy, or to implement a deep copy still needs to copy all the parameters. This pr is equivalent to using a partial copy, that is, copying the parameters that affect the display of the alarm information. The first question is related to whether to implement a full deep copy. If it is implemented, the ExecutorWrapper can be removed. If the solution is a partial deep copy, then it cannot be removed.

We can define a copy method in ExecutorWrapper, and then call it in BaseNotifyCtx, so that the change is small
```java
    @Override
    protected ExecutorWrapper clone() {
        ExecutorWrapper executorWrapper = null;
        try {
            executorWrapper = new ExecutorWrapper();
            executorWrapper.notifyItems = ImmutableList.copyOf(this.notifyItems);
            // TODO
            executorWrapper.executor = new CapturedExecutor(this.getExecutor());
        } catch (CloneNotSupportedException e) {
            // TODO
        }
        return executorWrapper;
    }
```

Reply:
I misunderstood before, this way will indeed be better. I will get it done as soon as possible.

Reply:
Alarm related information is in CapturedExecutor, I also suggest to remove the executorWrapper of BaseNotifyCtx.

Reply:
Made some changes. PTAL.

Reply:
LGTM
Thanks, can you add some UT later?

Reply:
> Made some changes. PTAL.

LGTM
Thanks, can you add some UT later?

Reply:
no problem.

Reply:
> > > Made some changes. PTAL.
> > 
> > 
> > LGTM Thanks, can you add some UT later?
> 
> no problem.

done.